src/main/java/org/jvnet/hudson/plugins/repositoryconnector/VersionParameterDefinition.java

```diff
@@ -53,35 +58,74 @@ public VersionParameterDefinition(String repoid, String groupid,
+    public Collection<Repository> getRepos(String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if(repoid.compareTo("ALL")==0 || repoid.compareTo(e.getKey())==0) {
+    			out.add(e.getValue());
+    		}
+    	}
+    	return out;
+    }
+    
+    
     @Exported
     public List<String> getChoices() {
-        Repository r = DESCRIPTOR.getRepo(repoid);
-        List<String> versionStrings = new ArrayList<String>();
-        if (r != null) {
-            File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
-            Aether aether = new Aether(DESCRIPTOR.getRepos(), localRepo);
-            try {
-                List<Version> versions = aether.resolveVersions(groupid, artifactid);
-                for (Version version : versions) {
-                    versionStrings.add(version.toString());
-                }
-            } catch (VersionRangeResolutionException ex) {
-                log.log(Level.SEVERE, "Could not determine versions", ex);
-            }
-            if (!versionStrings.isEmpty()) {
-                // reverseorder to have the latest versions on top of the list
-                Collections.reverse(versionStrings);
-                // add the default parameters
-                versionStrings.add(0, "LATEST");
-                versionStrings.add(0, "RELEASE");
+    	Collection<Repository> repos = getRepos(repoid);
+
+    	List<String> versionStrings = new ArrayList<String>();
+
+    	File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
+        log.info("VersionParameterDefinition: local repo "+localRepo.getAbsolutePath());
+        
+        Aether aether = new Aether(
+        		repos, localRepo, null, false, 
+        		RepositoryPolicy.UPDATE_POLICY_ALWAYS, 
+                RepositoryPolicy.CHECKSUM_POLICY_FAIL, 
+                RepositoryPolicy.UPDATE_POLICY_ALWAYS, 
+                RepositoryPolicy.CHECKSUM_POLICY_FAIL);
+        try {
+            List<Version> versions = aether.resolveVersions(groupid, artifactid);
+            while(versions.size()>0) {
+            	if(reverseOrder) {
+            		versionStrings.add(0,versions.remove(0).toString());
+            	} else {
+            		versionStrings.add(versions.remove(0).toString());
+            	}
             }
+        } catch (VersionRangeResolutionException ex) {
+            log.log(Level.SEVERE, "Could not determine versions", ex);
+        }
+        if(versionStrings.size()==0) {
+        	throw new RuntimeException("no versions found");
+        }
+        if (!versionStrings.isEmpty()) {
+            // reverseorder to have the latest versions on top of the list
+        	if(reverseOrder) {
+        		Collections.reverse(versionStrings);
+        	}
+            // add the default parameters
+        	// commented this because nexus is a bitch when it comes to dealing with 
```

Comment:
sorry, don't understand.
please also note, just adding a default in the config screen does not change anything for existing job configurations and these must stay functional, we can not expect a user to check and save all his jobs just because he updated this plugin.